pkg/expression/simple_rewriter.go

```diff
@@ -63,13 +63,20 @@ func ParseSimpleExpr(ctx BuildContext, exprStr string, opts ...BuildOption) (Exp
 
 // FindFieldName finds the column name from NameSlice.
 func FindFieldName(names types.NameSlice, astCol *ast.ColumnName) (int, error) {
-	dbName, tblName, colName := astCol.Schema, astCol.Table, astCol.Name
+	dbName, tblName, colName := astCol.Schema.L, astCol.Table.L, astCol.Name.L
 	idx := -1
+
 	for i, name := range names {
-		if !name.NotExplicitUsable && (dbName.L == "" || dbName.L == name.DBName.L) &&
-			(tblName.L == "" || tblName.L == name.TblName.L) &&
-			(colName.L == name.ColName.L) {
+		// Early continue if not usable or column name doesn't match
+		if name.NotExplicitUsable || name.ColName.L != colName {
+			continue
+		}
```

Comment:
Can we further simplify the code by putting db/table name match checking after this condition, so that the subsequent code won't need another `if` statement:
```
		// Boolean arithmetic will properly short-circuit the expression so column name checking would still be the most effective.
		if name.NotExplicitUsable || name.ColName.L != colName || !(dbName == "" || dbName == name.DBName.L) || !(tblName == "" || tblName == name.TblName.L) {
			continue
		}
```

Reply:
I prefer not to have such a lengthy boolean condition check. Thank you!

Reply:
OK. That sounds good too.